src/components/Recipes/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react'
+import { Link, useLocation } from 'react-router-dom'
+
+import { fetchRecipes } from 'api'
+import { IRecipe } from 'types'
+import Button from 'components/common/Button'
+
+import styles from './Recipes.module.scss'
+
+function Recipes() {
+  const [recipes, setRecipes] = useState([])
+  const location = useLocation()
+  const query = location.state.query
+
+  useEffect(() => {
+    fetchRecipes(query).then((response) => {
```

Comment:
you could try react-query for training, but if you want to leave it this way maybe please look at this https://stackoverflow.com/a/73190969